platforms/core-runtime/daemon-protocol/build.gradle.kts

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+plugins {
+    id("gradlebuild.distribution.api-java")
+}
+
+description = "The messages and types sent between client and daemon"
+
+dependencies {
+    api(project(":base-services"))
+    api(project(":logging-api"))
+    api(project(":serialization"))
+    api(project(":core"))
```

Comment:
Would be nice if we could avoid this dependency here. Since one of the stated goals of this PR is to allow different buckets of code to target different JVM versions, and since this code is used by the launcher, I think we should avoid pulling in core if possible. 

Do you know which code in project requires classes from core?

Reply:
I've added some comments.